File: backend/src/Locker.Backend.Domain/Entities/LockerSlot.cs
using Locker.Backend.Domain.Enums;

namespace Locker.Backend.Domain.Entities;

public class LockerSlot
{
    public int Index { get; set; }
    public LockerSlotStatus Status { get; set; } = LockerSlotStatus.Available;
    public string Size { get; set; } = "M";
    public string SensorState { get; set; } = "Closed";
    public string? BookingId { get; set; }
}

File: mobile/lib/features/locker/domain/entities/locker_slot.dart
import 'package:equatable/equatable.dart';

class LockerSlot extends Equatable {
  final int index;
  final int status; // Assuming 0 = available
  final String size;

  const LockerSlot({
    required this.index,
    required this.status,
    required this.size,
  });

  bool get isAvailable => status == 0;

  @override
  List<Object?> get props => [index, status, size];
}
